internal/testing/e2e/kind.go

```diff
@@ -227,19 +227,31 @@ func LoadKindDockerImage(clusterName string) env.Func {
 	return func(ctx context.Context, cfg *envconf.Config) (context.Context, error) {
 		log.Println("Loading docker image into kind cluster")
 
-		image := DockerImage(ctx)
-
-		var stdout, stderr bytes.Buffer
-		timeoutContext, cancel := context.WithTimeout(ctx, 20*time.Second)
-		defer cancel()
+		if err := loadImage(ctx, clusterName, DockerImage(ctx)); err != nil {
+			return nil, err
+		}
 
-		cmd := exec.CommandContext(timeoutContext, "kind", "load", "docker-image", image, image, "--name", clusterName)
-		cmd.Stderr = &stderr
-		cmd.Stdout = &stdout
-		if err := cmd.Run(); err != nil {
-			return nil, errors.New(stderr.String())
+		for _, image := range ExtraDockerImages() {
+			log.Printf("Loading additional docker image:%s into kind cluster", image)
+			if err := loadImage(ctx, clusterName, image); err != nil {
+				return nil, err
+			}
```

Comment:
Running `go test ./... -tags e2e -run TestHTTPMeshService` locally for me produces the following output, which looks like (at best) the same image is attempting to be loaded twice, but the e2e test also seems to consistently time out 😞 
```
2022/02/17 17:19:49 Loading docker image into kind cluster
2022/02/17 17:20:09 Loading additional docker image:consul-dev:ent into kind cluster
2022/02/17 17:20:29 Image: "consul-dev:ent" with ID "sha256:8492c3d4c6dcd4404bc0a20f232e5f7bac73f14c4f724cf1c7db9aabb03184c8" not yet present on node "consul-api-gateway-test-016573-control-plane", loading...
Image: "consul-dev:ent" with ID "sha256:8492c3d4c6dcd4404bc0a20f232e5f7bac73f14c4f724cf1c7db9aabb03184c8" not yet present on node "consul-api-gateway-test-016573-control-plane", loading...
```
I'm guessing with this code, specifying _both_ env vars isn't actually necessary?
```
E2E_APIGW_CONSUL_IMAGE=consul-dev:ent
E2E_APIGW_DOCKER_IMAGES=consul-dev:ent
```

Reply:
Welp, it looks like maybe the `-run` flag was obscuring the actual issue 🤦 

I modified `build-support/docker/Consul-Dev.dockerfile` to add a test license 
```
COPY test.lic test.lic
RUN chmod +r test.lic
ENV CONSUL_LICENSE_PATH=./test.lic
```
Which _works fine_ when running `docker run -t --rm consul-dev:ent` but I'm guessing maybe it isn't being copied/set in the `consul-api-gateway` image we're building. Would be fine to add extra tooling to support the explicit license file in a folllowup PR.

Reply:
yeah, so I believe Nick did some work to set up the license mounting in our test setu to work with a `CONSUL_LICENSE` env variable that gets leveraged in CI, see:
 https://github.com/hashicorp/consul-api-gateway/blob/dc7562ac7226e582e162c3b0a548163e5715283a/internal/testing/e2e/consul.go#L30

I'm thinking you can probably just `export CONSUL_LICENSE="$(cat ./test.lic)"`.

RE the environment variables, you _will_ need both -- the first is the Consul image to run in the container deployed in the testing `kind` cluster (which should be your locally built image), and the `E2E_APIGW_DOCKER_IMAGES` is a little shim that does a `kind load` of said image prior to deploying Consul.

Reply:
Line 230 is loading up a docker image of the current codebase so that it can be deployed, it's not controlled by any env variables, and always needs to happen. This other bit is to load up any additional images that your environment may need.